test/test_torch.py

```diff
@@ -14109,6 +14109,27 @@ def test_helper(shape, numel, memory_format, device):
         test_helper((10, 3, 32, 32), 10 * 3 * 32 * 32, torch.channels_last, device)
         test_helper((3, 10, 3, 32, 32), 3 * 10 * 3 * 32 * 32, torch.channels_last_3d, device)
 
+    def test_memory_format_proparation_rules(self, device):
+        contiguous = torch.rand(10, 3, 5, 5, device=device) 
+        cl = torch.rand(10, 3, 5, 5, device=device).contiguous(memory_format=torch.channels_last)
+        ambiguous = torch.rand(10, 3, 1, 1, device=device).contiguous(memory_format=torch.channels_last)
```

Comment:
I'm afraid this test isn't testing what it thinks it is testing. `.contiguous(memory_format)` is a light call that does not force-reset the strides, what you need here is `.to(memory_format)`:
```
In [4]: ambiguous = torch.rand(10, 3, 1, 1, device="cuda").contiguous(memory_format=torch.channels_last)                                                                                                                

In [5]: ambiguous.stride()                                                                                                                                                                                              
Out[5]: (3, 1, 1, 1)

In [6]: ambiguous = torch.rand(10, 3, 1, 1, device="cuda").to(memory_format=torch.channels_last)                                                                                                                        

In [7]: ambiguous.stride()                                                                                                                                                                                              
Out[7]: (3, 1, 3, 3)
```
For the latter case (even though it's still ambiguous) `.suggest_memory_format` will return `channels_last`, for the former it will return `contiguous`.

Reply:
I verified that the test fails if I replace `.contiguous` here with `.to`

Reply:
Thanks! Fixed.